### PR TITLE
Implement Product permissions

### PR DESCRIPTION
**Pull request by m4v15**
- [x] add permissioning on routes
- [x] fix the tests that break because of that
- [x] add a few more permissioning tests
- [x] update the docs

### CODE REVIEW

File: tests/controllers/product.test.js [hunk 1]
```diff
@@ -113,80 +123,222 @@ tape('POST /products with invalid product data', t => {
...
-              dropCollectionAndEnd(Product, t)
-            })
-            .catch(err => {
-              t.fail(err)
-              dropCollectionAndEnd(Product, t)
-            })
-        })
-    })
-    .catch(err => t.end(err))
+tape('PUT /products/:id with id of something not in the database, logged in via Oauth as SUPER', t => {
```
Comment (mattlub) [inline tests/controllers/product.test.js:142]: oauth! :+1:

File: tests/controllers/product.test.js [hunk 2]
```diff
@@ -113,80 +123,222 @@ tape('POST /products with invalid product data', t => {
...
+        dropCollectionsAndEnd([Product, User], t)
+      })
+      .catch(err => {
+        t.fail(err)
+        dropCollectionsAndEnd([Product, User], t)
+      })
+    })
+    .catch(err => t.end(err))
+  })
+  .catch(err => t.end(err))
```
Comment (mattlub) [inline tests/controllers/product.test.js]: you could flatten all this to just one `catch`, it looks like all you need is the `productToBeDeleted` variable to be defined in the outermost scope

File: tests/index.test.js [hunk 3]
```diff
@@ -14,6 +15,7 @@ dbConnection.once('open', () => {
     .then(() => Place.remove({}))
     .then(() => Event.remove({}))
     .then(() => Product.remove({}))
+    .then(() => Token.remove({}))
```
Comment (mattlub) [inline tests/index.test.js]: Promise.all ?

**Review by mattlub — CHANGES_REQUESTED** [review]

**Comment by mattlub** [comment]
@m4v15 nice one :ear_of_rice:

File: tests/controllers/product.test.js (hunk 2 above)
Comment (m4v15) [reply tests/controllers/product.test.js]: yeah so it got complicated when accessing the id to compare to or something, I need to think it through more - I tried manually setting the id in the fixtures file but that threw a load of errors for some reason

File: tests/index.test.js (hunk 3 above)
Comment (m4v15) [reply tests/index.test.js]: yeah, this was in the middle of manic debugging so I left it for later, but yeah you're right

**Review by mattlub — APPROVED** [review]

**Review by des-des — APPROVED** [review]

**Pull request merged**